src/extent.c

```diff
@@ -438,11 +448,22 @@ extents_first_fit_locked(tsdn_t *tsdn, arena_t *arena, extents_t *extents,
 	    &extents_bitmap_info, (size_t)pind); i < NPSIZES+1; i =
 	    (pszind_t)bitmap_ffu(extents->bitmap, &extents_bitmap_info,
 	    (size_t)i+1)) {
-		assert(!extent_heap_empty(&extents->heaps[i]));
+		mutex_pool_lock(tsdn, &arena->extents_mutex_pool, i);
 		extent_t *extent = extent_heap_first(&extents->heaps[i]);
+		if (!extent) {
+			mutex_pool_unlock(tsdn, &arena->extents_mutex_pool, i);
+			continue;
+		}
 		assert(extent_size_get(extent) >= size);
 		if (ret == NULL || extent_snad_comp(extent, ret) < 0) {
+			extent_activate_locked(tsdn, arena, extents, extent);
+			mutex_pool_unlock(tsdn, &arena->extents_mutex_pool, i);
+			if (ret) {
```

Comment:
nit: if (ret != NULL)

also maybe some comments about why activate + deactivate is needed?